File: src/DRandomFoundational.cs
/*******************************************************************************
*  Copyright by the contributors to the Dafny Project
*  SPDX-License-Identifier: MIT
*******************************************************************************/

using System;

namespace DafnyLibraries {
  public partial class DRandomFoundational : BaseInterface.Base, UniformImplementation.Unif, BernoulliImplementation.Bernoulli, GeometricImplementation.Geometric, UniformImplementation.UniformFoundational, BernoulliExpNegImplementation.BernoulliExpNeg, DiscreteGaussianImplementation.DiscreteGaussian, DiscreteLaplaceImplementation.DiscreteLaplace {
    private Random r;

    public DRandomFoundational() {
      this.r = new Random();
    }

    public bool Coin() {
      return this.r.Next(2) == 0;
    }
  }
}

File: src/DRandomFoundational.java
/*******************************************************************************
*  Copyright by the contributors to the Dafny Project
*  SPDX-License-Identifier: MIT
*******************************************************************************/

package DafnyLibraries;

import java.security.SecureRandom;
import java.math.BigInteger;

public class DRandomFoundational extends _ExternBase_DRandom {
  private SecureRandom s;

  public DRandomFoundational() {
    this.s = new SecureRandom();
  }

  @Override
  public boolean Coin() {
    return s.nextBoolean();
  }
}
